### ISSUE TITLE
linter: eslint-plugin-jsdoc(check-tag-names) gives false positive when `@` is used inside `[`,`]`

### ISSUE THREAD

**Issue by TheAlexLichter** [oxc-project/oxc#10910]
### What version of Oxlint are you using?

0.16.9

### What command did you run?

pnpm oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
   "plugins": [
    "jsdoc"
  ],
  "rules": {
    "jsdoc/check-tag-names": [
      "error",
      {
        "definedTags": []
      }
    ]
  }
}
```

Or any content in `definedTags`

### What happened?

~Using `definedTags` should be a list in addition to the existing tags that are valid. When now using an @, e.g. in the description of `@see`, it will throw. This does not happen when not using `definedTags`~

oxlint is parsing `@parcel/watcher](https://github.com/parcel-bundler/watcher)` as a jsdoc tag, when it is not one, as `@parcel/watcher](https://github.com/parcel-bundler/watcher)` is not in the allowed list of tag names, `jsdoc/check-tag-names` reports a diagnostic.

Example:

```ts
export interface ConfigSchema {
  /**
   * @see [@parcel/watcher](https://github.com/parcel-bundler/watcher)
   */
  watcher: 'chokidar' | 'parcel' | 'chokidar-granular'
}
```

<img width="882" alt="Image" src="https://github.com/user-attachments/assets/55e55bfd-a2da-4504-9dad-9f29f47f3b56" />
